controller/appcontroller.go

```diff
@@ -993,6 +994,11 @@ func (ctrl *ApplicationController) watchSettings(ctx context.Context) {
 				ctrl.stateCache.Invalidate()
 				prevResourceInclusions = newSettings.ResourceInclusions
 			}
+			if !reflect.DeepEqual(prevConfigManagementPlugins, newSettings.ConfigManagementPlugins) {
```

Comment:
Added as it allows me to test. I could not get `kubectl patch cm argocd-cm` to work because our values are YAML and it just makes a big mess.